cipd_packages/device_doctor/lib/src/android_device.dart

```diff
@@ -380,7 +381,68 @@ class AndroidDevice implements Device {
 
   @override
   Future<void> recover() async {
-    await eval('adb', <String>['-s', deviceId!, 'reboot'], canFail: false);
+    await cleanDevice();
+  }
+
+  @visibleForTesting
+  Future<void> deletePackageCache() async {
+    final ProcessResult result = Process.runSync('adb', <String>['shell', 'pm', 'list', 'packages']);
+
+    if (result.exitCode != 0) {
+      throw Exception(result.stderr as String);
+    }
+    final packages = <String>[];
+
+    // Listen to the stdout and stderr streams
+    LineSplitter().convert(result.stdout).forEach((data) {
+      final packageMatch = RegExp(r'package:(.+)$').firstMatch(data);
+      if (packageMatch != null) {
+        final packageName = packageMatch.group(1);
+        if (packageName != null) {
+          packages.add(packageName);
+        }
+      }
+    });
+    for (final p in packages) {
+      Process.runSync('adb', <String>['shell', 'pm', 'clear', p]);
```

Comment:
should we check the output? even if we don't fail on non-zero, we could at least leave a debug print.

Reply:
done